File: Quote.py
from dataclasses import dataclass


@dataclass
class Quote:
    author: str
    quote: str
    tags: list[str]
    created_at: str
    updated_at: str

    def __init__(self, author: str, quote: str, created_at: str, updated_at: str, tags: list[str] | None = None):
        self.author = author
        self.quote = quote
        self.tags = tags if tags is not None else []
        self.created_at = created_at
        self.updated_at = updated_at

File: web/src/quote-utilities/quote.js
export class Quote {
    static author = "";
    static quote = "";
    static tags = [];
    static createdAt = "";
    static updatedAt = "";

    constructor(author, quote, tags, createdAt, updatedAt) {
        this.author = author;
        this.quote = quote;
        this.tags = tags;
        this.createdAt = createdAt;
        this.updatedAt = updatedAt;
    }


    get authorName() {
        return this.author;
    }

    get quoteText() {
        return this.quote;
    }

    get quoteTags() {
        return this.tags;
    }

    get createdAtDate() {
        return this.createdAt;
    }

    get updatedAtDate() {
        return this.updatedAt;
    }

    tagsToString() {
        return this.tags.join(', ');
    }
}